traits/traits_listener.py

```diff
@@ -1195,22 +1195,15 @@ def owner_deleted(self, ref):
 
 class ListenerHandler(object):
     def __init__(self, handler):
-        if type(handler) is MethodType:
```

Comment:
I can't see a good reason to do an exact type check rather than the usual `isinstance` check here.